### PR TITLE
Added DELETE encounters endpoint

### PR DESCRIPTION
**Pull request by sher812**
Added Delete Encounters Endpoint:

Deletes specified encounter (using objectID). Also removes encounters from related persons/users.

Manually tested endpoints with postman.

NOTE: Has an identical method deleteUserEncounter() in user.services as with the delete persons endpoint (#55). 

Fixed(#54)

People who worked on this issue:
@UoA-sjer215 and @sher812

### CODE REVIEW

File: backend/src/routes/encounter.route.ts [hunk 1]
```diff
@@ -2,12 +2,13 @@
...
  import { Router } from 'express';
- import {createEncounter, updateEncounter} from '../controllers/encounter.controller';
+ import {createEncounter, deleteEncounters, updateEncounter} from '../controllers/encounter.controller';
  
  const routes = Router();
  
  routes.post('/', createEncounter);
- routes.put('/:id', updateEncounter)
+ routes.put('/:id', updateEncounter);
+ routes.delete(':/id', deleteEncounters)
```
Comment (butterplup) [inline backend/src/routes/encounter.route.ts:11]: is this meant to be `:/id` for the `routes.delete`?

Comment (AdamAWiener) [reply backend/src/routes/encounter.route.ts:11]: Yeah I think it should be `/:id` right?

Comment (sher812) [reply backend/src/routes/encounter.route.ts]: Ah whoops yeah

**Review by butterplup — DISMISSED** [review]
i think this looks good

**Comment by sher812** [comment]
Resolved merge conflicts

**Review by butterplup — APPROVED** [review]
yeah LGTM 🥸

**Review by hajineats — APPROVED** [review]
LGTM

**Review by jimwang6012 — DISMISSED** [review]
LGTM!

**Review by butterplup — APPROVED** [review]
LGTM

**Comment by sdiv877** [comment]
LGTM! 👍

**Review by sdiv877 — APPROVED** [review]

**Review by sdiv877 — DISMISSED** [review]
LGTM! Sorry about the merge conflicts 😅

**Review by lin8231 — COMMENTED** [review]
Looking great! Just have a quick question :)

File: backend/src/controllers/encounter.controller.ts [hunk 2]
```diff
@@ -142,6 +142,39 @@ export const getEncounter = async (
...
+    const current_user = await userService.getUserByAuthId(auth_id);
+    const id = req.params.id;
+    const user_encounters = current_user?.encounters;
+    let string_encounters = user_encounters?.map(x => x.toString());
+
+    if (string_encounters?.includes(id.toString())) {
+      try {
+        // Delete user from database
+        await encounterService.deleteEncounter(req.params.id);
+        await personService.deletePersonEncounters(req.params.id);
```
[the remark this answers is not in the document]
Comment (sher812) [reply backend/src/controllers/encounter.controller.ts:162]: Ah okay so in other words, did you want to me to make sure that the services on 161, 162  and 163 don't fail before I send out an HTTP OK?

**Review by schi314 — DISMISSED** [review]

File: backend/src/controllers/encounter.controller.ts [hunk 3]
```diff
@@ -142,6 +142,44 @@ export const getEncounter = async (
...
+    
+    const current_user = await userService.getUserByAuthId(auth_id);
+    const id = req.params.id;
+    const user_encounters = current_user?.encounters;
+    let string_encounters = user_encounters?.map(x => x.toString());
+
+    if (string_encounters?.includes(id.toString())) {
+      try {
+        // Delete encounter from Encounter and Person collection
+        const deleteEncounterResult = await encounterService.deleteEncounter(req.params.id);
```
Comment (AdamAWiener) [inline backend/src/controllers/encounter.controller.ts:161]: Can we just change these to the `id` declared above?

File: backend/src/controllers/encounter.controller.ts [hunk 4]
```diff
@@ -142,6 +142,44 @@ export const getEncounter = async (
...
+        const deletePersonEncountersResult = await personService.deletePersonEncounters(req.params.id);
+        const deleteUserEncounterResult = await userService.deleteUserEncounter(req.params.id);
+
+        // Notify frontend that the operation was successful
+        if (deleteEncounterResult && deletePersonEncountersResult && deleteUserEncounterResult) {
+          res.sendStatus(httpStatus.OK).end();
+        } else {
+          res.sendStatus(httpStatus.BAD_REQUEST);
```
Comment (AdamAWiener) [inline backend/src/controllers/encounter.controller.ts:169]: Needs a `.end()` after the response status.

**Review by butterplup — APPROVED** [review]

File: backend/src/controllers/encounter.controller.ts [hunk 5]
```diff
@@ -142,6 +142,44 @@ export const getEncounter = async (
...
+    if (string_encounters?.includes(id.toString())) {
+      try {
+        // Delete encounter from Encounter and Person collection
+        const deleteEncounterResult = await encounterService.deleteEncounter(req.params.id);
+        const deletePersonEncountersResult = await personService.deletePersonEncounters(req.params.id);
+        const deleteUserEncounterResult = await userService.deleteUserEncounter(req.params.id);
+
+        // Notify frontend that the operation was successful
+        if (deleteEncounterResult && deletePersonEncountersResult && deleteUserEncounterResult) {
+          res.sendStatus(httpStatus.OK).end();
```
Comment (jimwang6012) [inline backend/src/controllers/encounter.controller.ts:167]: Potentially `204 NO CONTENT` for successful delete operation?

File: backend/src/services/encounter.service.ts [hunk 6]
```diff
@@ -47,7 +47,16 @@ const deleteEncounterPerson = async (personID: string) => {
 }
 
 const deleteEncounter = async (encounterID: String) => {
-  await Encounter.deleteOne({ _id: encounterID }).exec();
+  const result = await Encounter.deleteOne({ _id: encounterID }).exec();
+  console.log("RESULT: ", result);
```
Comment (AdamAWiener) [inline backend/src/services/encounter.service.ts:51]: Do we still need this log?

**Review by AdamAWiener — COMMENTED** [review]
Looks good, just a few things to clean up!

File: backend/src/controllers/encounter.controller.ts (hunk 3 above)
Comment (sher812) [reply backend/src/controllers/encounter.controller.ts]: Fixed

File: backend/src/services/encounter.service.ts (hunk 6 above)
Comment (sher812) [reply backend/src/services/encounter.service.ts]: fixed

File: backend/src/controllers/encounter.controller.ts (hunk 4 above)
Comment (sher812) [reply backend/src/controllers/encounter.controller.ts]: Fixed

**Review by AdamAWiener — APPROVED** [review]
LGTM, thanks for fixing those issues I brought up!

**Review by jimwang6012 — APPROVED** [review]
LGTM!

File: backend/src/controllers/encounter.controller.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (sher812) [reply backend/src/controllers/encounter.controller.ts]: Fixed

File: backend/src/controllers/encounter.controller.ts (hunk 5 above)
Comment (sher812) [reply backend/src/controllers/encounter.controller.ts:167]: Fixed

**Pull request merged**